### ISSUE TITLE
Polygons and polylines are rendered without an 'order' value

### ISSUE THREAD

**Issue by matteblair** [tangrams/tangram#503]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

Load this scene:
```yaml
scene:
    background:
        color: lightblue
sources:
    mapzen:
        type: TopoJSON
        url: https://tile.mapzen.com/mapzen/vector/v1/all/{z}/{x}/{y}.topojson
layers:
    earth:
        data:
            source: mapzen
        draw:
            polygons:
                color: grey
            lines:
                width: 2px
                color: yellow
```

**RESULT:**

Yellow polylines are drawn around grey Earth layer polygons.

**EXPECTED RESULT:**

The `lines` and `polygons` draw rules on the `earth` layer have no `order` values specified, and `order` is a required parameter for these styles, as noted in the documentation (https://mapzen.com/documentation/tangram/draw/#order), so both the yellow polylines and the grey polygons should _not_ be drawn.

**ENVIRONMENT:**

MacOS 10.11.6, Firefox 51.0.1, Tangram Play v0.5.2, Tangram v0.11.8

**OTHER:**

This differs from the current behavior of tangram-es, as noted here: https://github.com/tangrams/tangram-es/issues/1314

**Comment by dmvaldman** [comment]
I would have assumed that `order` would not be a required parameter, and that it would default to something (even if it's essentially random), as it may cause friction to a newcomer.

**Comment by bcamper** [comment]
It's hard to make it default to anything useful. If it defaults to 0 (which
is probably what's happening, even if unintentional?), everything ends up
intersecting and draw order will be fairly random. I suppose a more useful
default could be to auto-increment order as each layer in the `layers`
block is processed. However, layers can be nested, so not sure how that
gets applied, and technically speaking the YAML mapping itself is not
ordered (I think in practice it may be possible to access the order but I
think it's not guaranteed). It's also difficult to apply auto-order in any
kind of scene `import`/merge scenario. Overall it's unclear if many useful
maps can be made without specifying order? I'm open to ideas though if this
can be improved! Render order has always been a challenging concept in
Tangram.

**Comment by matteblair** [comment]
@dmvaldman That was actually the original behavior of both libraries, but this proved unsustainable when we started trying to show the same styling consistently across JS and ES. 

The problem is that when `order` has a default value, the relative layering of all features using that default `order` value is _totally arbitrary_. Those features may be layered consistently in the one browser where you write your scene, and you may think that you've laid out everything correctly, but it only looks that way by chance. So when you put the same scene file onto a phone things will be layered in a different, but _totally valid_ order. This makes you :(

**Comment by dmvaldman** [comment]
I lean on the side of being generous to the user who may not be following the directions very carefully. I prefer loose input with unreliable output to strict input with reliable output. Of course when input is *wrong* Tangram should break. I don't think this qualifies for that.

In the web-world, depth order is not well-defined between `translateZ()`, DOM-element-order in "flat" contexts, but browsers still draw stuff, even if they disagree on how. Enough for me to write [bug reports](https://bugs.chromium.org/p/chromium/issues/detail?id=573908) about it.

**Comment by matteblair** [comment]
I understand that browsers may tolerate ambiguity like this, but let us acknowledge that the web is not a paragon of pleasant developer experience in this regard :)

I think a good goal for this case is to be strict with input and provide good feedback and tools. The feedback and tools still need work on our end.

**Comment by bcamper** [comment]
Features without an `order` will no longer be rendered (**except when the blend mode is `overlay`**, since I don't believe `order` should be required in that case and has no rendering effect), and a warning will be logged once per layer that this is encountered. c2ad34bf127489b267d1d6c46cbce4b0bab9b9c4

**Comment by bcamper** [comment]
Released in [v0.12.0](https://github.com/tangrams/tangram/releases/tag/v0.12.0)
